Добавляет совет по инициализации состояния (#5328)
doka-guide/content: recipes/expressive-react-maze/practice/vitya-ne.md

> ☝️ Обратите внимание как устанавливается первоначальное значение состояния в компоненте Maze.
> Аргументом хука `useState` тут является функция, возвращающая результат выполнения `Array.from()`:

Reviewer: ```suggestion
🛠 Обратите внимание как устанавливается первоначальное значение состояния в компоненте `Maze`. Аргументом хука `useState` тут является функция, возвращающая результат выполнения `Array.from()`:
```

```tsx

> // пример A

Reviewer: ```suggestion
// Пример A
```

const [walls, setWalls] = useState<Wall[]>(
  () => Array.from({ length: fieldSize }, () => ({
    left: 0,
    width: 0,
  }))
)
```

> Всё будет работать точно также, если использовать `Array.from()` непосредственно как аргумент `useState`:

Reviewer: ```suggestion
Всё будет работать точно так же, если использовать `Array.from()` непосредственно как аргумент `useState`:
```

```tsx

> // пример B

Reviewer: ```suggestion
// Пример B
```

const [walls, setWalls] = useState<Wall[]>(
  Array.from({ length: fieldSize }, () => ({
    left: 0,
    width: 0,
  }))
)
```

> В чём же разница между _примером A_ и _примером B_ ?

Reviewer: ```suggestion
В чём же разница между _примером A_ и _примером B_?
```

> Представим, что установка первоначального значения состояния требует вычислений и затратно по времени.

Reviewer: ```suggestion
Представим, что установка первоначального значения состояния требует вычислений и затратна по времени.
```

> Инициализация состояния происходит только при первом рендере компонента.
> Функция `() => Array.from(..` в _примере A_ будет выполнена только один раз.
> А выполнение `Array.from()` в _примере B_ будет происходить при каждом рендере компонента.

Reviewer: ```suggestion
Инициализация состояния происходит только при первом рендеринге компонента. Функция `() => Array.from()` в _примере A_ будет выполнена только один раз. Выполнение `Array.from()` в _примере B_ будет происходить при каждом рендеринге компонента.
```

> 💡 Возможность передавать в качестве аргумента `useState` функцию предусмотрена для случаев когда вычисление первоначального состояния мы хотим выполнить только один раз.

Reviewer: ```suggestion
💡 Возможность передавать в качестве аргумента функцию  `useState` предусмотрена для случаев, когда вычисление первоначального состояния хотим выполнить только один раз.
```

Author: вот тут не соглашусь: `useState` работает как функция и может принимать аргумент. Мой совет как раз и описывает разницу между передачей значения и функции в качестве этого аргумента.
тоесть функция useState принимает другую функцию как аргумент

Author: @TatianaFokina, немного переделал
